src/index.js

```diff
@@ -0,0 +1,254 @@
+    if (column === row) {
+      return 0;
+    }
+
+    if (column + row + 1 === this.board.dimension) {
+      return this.board.dimension - 1;
+    }
+
+    return null;
+  };
+
+  Evaluator.prototype.isOnDiagonal = function () {
+    const columnIndex = this.getColumn();
+    const rowIndex = this.getRow();
+
+    return (
+      columnIndex === rowIndex
+      || columnIndex + rowIndex + 1 === this.board.dimension
+    );
+  };
+
+  return Evaluator;
+}());
+
+const Board = (function () {
+  function Board(dimension = 3) {
+    this.dimension = dimension;
+    this.board = new Array(dimension * dimension).fill('');
+  }
+
+  Board.prototype.get = function (index) {
+    return this.board[index];
+  };
+
+  Board.prototype.put = function (item, index) {
+    this.board[index] = item;
+  };
+
+  Board.prototype.isFilled = function () {
+    return this.board.filter(e => e).length === (this.dimension * this.dimension);
+  };
+
+  return Board;
+}());
+
+const Game = (function () {
+  function Game(board) {
+    this.board = board;
+    this.playerX = null;
+    this.playerO = null;
+    this.currentPlayer = null;
+    this.finished = false;
+  }
+
+  Game.prototype.render = function () {
+    const cells = document.querySelectorAll('.cell');
+    cells.forEach((el, i) => {
+      el.textContent = this.board.get(i);
+    });
+  };
+
+  function hideAnnouncement() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'none';
+  }
+
+  function displayWinner() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'block';
+    announcementNode.textContent = `${this.currentPlayer.name} wins!`;
+  }
+
+  function displayDraw() {
+    const announcementNode = document.querySelector('#announcement');
+    announcementNode.style.display = 'block';
+    announcementNode.textContent = 'Draw!';
+  }
+
+  Game.prototype.run = function () {
+    const game = this;
+    game.bootstrap();
+
+    const cells = document.querySelectorAll('.cell');
+    cells.forEach((el) => {
+      el.addEventListener('click', (e) => {
+        const index = e.target.dataset.index - 1;
+        if (!game.board.get(index) && !game.finished) {
+          game.board.put(game.currentPlayer.symbol, index);
+          game.render();
+          const winning = new Evaluator(game.board, index + 1).isWinning();
+          if (winning) {
+            game.currentPlayer.score += 1;
+            game.displayPlayerInfo();
+            displayWinner.bind(game)();
+            game.finished = true;
+          } else if (game.board.isFilled()) {
+            displayDraw();
+          } else {
+            game.switchPlayers();
+          }
+        }
+      });
+    });
+
+    const settingBtn = document.querySelector('#settings');
+    settingBtn.addEventListener('click', () => {
+      const modal = document.querySelector('#modal');
+      modal.style.display = 'block';
+    });
```

Comment:
Let's separate the DOM from application logic, you can create another file for DOM.

![image](https://user-images.githubusercontent.com/31897434/73420341-70e18b80-4322-11ea-840a-fa14e04b592b.png)

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/

Reply:
Resolved